### DISCUSSION TITLE
BLOB (and TEXT) length limits on MySQL?

### DISCUSSION THREAD

**Discussion by aaronkavlie-wf** [sqlalchemy/sqlalchemy#11893] [Usage Questions] [answered]
`types.BLOB` accepts a length argument, and the documentation [indicates](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.BLOB.params.length) that MySQL supports a length on this field. It also indicates [here](https://docs.sqlalchemy.org/en/20/dialects/mysql.html#index-length) that length is supported on `BLOB`, as well as `TEXT`

However, [MySQL docs](https://dev.mysql.com/doc/refman/8.4/en/blob.html) do not mention support for a length on `BLOB` fields. It's a fixed capacity of 65,535 bytes (there are a few other variants such as `MEDIUMBLOB` for smaller/larger capacities).

In my testing, Alembic passed a length I set to modify a column type:

```sql
ALTER TABLE tasks MODIFY foo BLOB(6000) NULL;
```

However, the length was ignored, and there's no way to set a length on that field.

Why do the SQLAlchemy docs say length is supported here?

**Comment by CaselIT** [comment, marked as answer]
Hi,

Mysql docs seems to be quite poor. It's specified here: https://dev.mysql.com/doc/refman/8.4/en/string-type-syntax.html

> An optional length M can be given for this type. If this is done, MySQL creates the column as the smallest [BLOB](https://dev.mysql.com/doc/refman/8.4/en/blob.html) type large enough to hold values M bytes long.

Same for text

> An optional length M can be given for this type. If this is done, MySQL creates the column as the smallest [TEXT](https://dev.mysql.com/doc/refman/8.4/en/blob.html) type large enough to hold values M characters long.

**Reply by aaronkavlie-wf** [reply]
Oh, interesting.
I just modified my migration to specify a length of 70,000 (just a bit over the threshold for the standard `BLOB` type).
Sure enough, it created the table with a `MEDIUMBLOB` type for that column.

Thanks for the explanation!

**Reply by CaselIT** [reply]
no problem, mysql should really do a better job of documenting the types
